File: application/src/main/java/com/github/cb0s/gs9dof/messages/ConnectionDetails.java
package com.github.cb0s.gs9dof.messages;

import com.fasterxml.jackson.annotation.JsonProperty;
import de.wuespace.telestion.api.message.JsonMessage;

public record ConnectionDetails(
		@JsonProperty
		String com,
		@JsonProperty
		int baudRate,
		@JsonProperty
		boolean connected) implements JsonMessage {
}

File: client/src/model/messages/receive/connection-details.ts
import { JsonSerializable } from '@wuespace/telestion-client-types';
import { ClassNamePrefix } from '../../channels';

export interface ConnectionDetails extends Record<string, JsonSerializable> {
	com: string;
	baudRate: number;
	connected: boolean;
	className: `${ClassNamePrefix}.ConnectionDetails`;
}
